File: baekJoon/1613.py
from sys import stdin

n,k = map(int, stdin.readline().split())

graph, ans = [[] for _ in range(n)], [[0 for _ in range (n)] for _ in range(n)]

for _ in range(k):
    i,j = map(int, stdin.readline().split())
    graph[i-1].append(j-1)

visit = [0 for _ in range(n)]
def update(start, i):
    visit[i] = 1
    if graph[i] == [] :
        return

    for j in graph[i]:
        ans[start][j] = -1
        # ans[i][j] = -1
        ans[j][start] = 1
        # ans[j][i] = 1
        update(start, j)

def upd(before, i):
    if visit[i]:
        return
    visit[i] = 1
    # print(before)

    for j in graph[i]:
        for b in before:
            ans[b][j] = -1
            ans[j][b] = 1
        upd(before+[j], j)

for i in range(n):
    if visit[i] == 0:
        upd([i], i)
        # print(visit)

for tmp in ans:
    print(tmp)


it = int(stdin.readline())
for _ in range(it):
    i,j = map(int, stdin.readline().split())
    print(ans[i-1][j-1])

File: baekJoon/1613.cpp
#include <iostream>
#include <vector>

using namespace std;
int ans[400][400] = {0,};
vector<int>  graph[400];
int visit[400] = {0,};

void update(int start, int i){
    for(vector<int>::iterator it= graph[i].begin(); it != graph[i].end(); it++){
        ans[start][*it] = -1;
        ans[*it][start] = 1;
        update(start, *it);
    }
}


int main(){
    ios::sync_with_stdio(false);
    int n, k, s, e, num;
    cin >> n >> k;

    for(int i=0; i<k; i++){
        cin >> s >> e;
        graph[s-1].push_back(e-1);
    }

    for(int i=0; i<n; i++)
        update(i,i);

    cin >> num;

    for(int i=0; i<num; i++){
        cin >> s >> e;
        cout << ans[s-1][e-1]<< '\n';
    }

    return 0;
}


int arr[] = {0, 1, 3, 5, 7};

int& f(int n){
    return arr[n];
}

1) f(0) = 100;
2) f(0) = 1 + f(1) + f(2);
3) int v = f(2); v++;